test/stp_card_params_test.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:flutter_stripe/flutter_stripe.dart';
+import 'package:test/test.dart';
+
+STPCardParams cardParams() {
+  final now = new DateTime.now();
+  return new STPCardParams(
+    "4242424242424242",
+    now.month,
+    now.year,
+    cvc: "123",
+    address: new STPAddress(postalCode: "12345")
+  );
+}
+
+void main() {
+  cardTests();
+  expMonthTests();
+  cvcTests();
+  zipTests();
+}
+
+void cardTests() {
+  test('valid card data is valid', () {
+    var card = cardParams();
+    final actual = card.isValid();
+    expect(actual, true);
+  });
+
+  test('short card number is invalid', () {
+    var card = cardParams();
+    card.number = "424242424242";
+    final actual = card.isValid();
+    expect(actual, false);
+  });
+
+  test('long card number is invalid', () {
+    var card = cardParams();
+    // valid luhn number
+    card.number = "42424242424242426";
+    final actual = card.isValid();
+    expect(actual, false);
+  });
+}
+
+void expMonthTests() {
+  test('zero month is invalid', () {
+    var card = cardParams();
+    card.expMonth = 0;
+    expect(card.isValid(), false);
+  });
+
+  test('negative month is invalid', () {
+    var card = cardParams();
+    card.expMonth = -1;
+    expect(card.isValid(), false);
+  });
+
+  test('current month is valid', () {
+    var card = cardParams();
+    expect(card.isValid(), true);
+  });
+
+  test('earlier month than now is invalid', () {
+    var card = cardParams();
+    final previousMonth = (card.expMonth + 10) % 12 + 1;
+    if (previousMonth > card.expMonth) {
+      card.expYear -= 1;
+    }
+    card.expMonth = previousMonth;
+    expect(card.isValid(), false);
+  });
+
+  test('earlier year than now is invalid', () {
+    var card = cardParams();
+    card.expYear -= 1;
+    expect(card.isValid(), false);
+  });
+}
+
+void cvcTests() {
+  test('null cvc is valid when not required', () {
+    var card = cardParams();
+    card.cvc = null;
+    expect(card.isValid(cvcRequired: false), true);
```

Comment:
If the above should return true in the case of an invalid cvc, but required = false, then should probably a test for both that and a zip that's invalid, but not required.